as_ikea.py

```diff
@@ -1,16 +1,50 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import random
+
 """Make your last name sound as an Ikea product.
 
 As seen in https://www.facebook.com/max.abelev/posts/3347430315294332
 """
 
 def as_ikea(name: str) -> str:
-    mapper = dict(a="ä", o="ö", u="ü")
-    for letter in mapper.keys():
-       name = name.replace(letter, mapper[letter])
-    return name[::-1]
+    """
+    Convert letters to ikea letters
+    :param: string
+    :return: ikea string
+    """
+    mapper = dict(a=["à", "á", "â", "ä", "æ", "ã", "ā"],
+                  o=["ô", "ö", "ò", "ó", "œ", "ø", "ō"],
+                  u=["û", "ü", "ù", "ú", "ū"],
+                  e=["è", "é", "ê", "ë", "ē", "ė"])
+
+    for index, letter in enumerate(name.lower(), start=0):
+        if mapper.get(letter):
+            name = replacer(name, random.choice(mapper[letter]), index)
+
+    return name[::-1].title()
+
+
+def replacer(s: str, newstring: str, index: int, nofail: bool = False) -> str:
+    """
```

Comment:
Is intent to replace just one occurrence of the letter here? 
I would eliminate a flag here for alternative behavior, if we wanted branching perhaps two functions is beyter